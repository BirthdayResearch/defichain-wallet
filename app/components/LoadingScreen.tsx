import * as React from 'react'
import { View } from 'react-native'
import Spinner from 'react-native-loading-spinner-overlay'
import { tailwind } from '../tailwind'
import { translate } from '../translations'

export default function LoadingScreen (props: { message?: string }): JSX.Element {
  const { message } = props
  const loadingMessage = message ?? translate('screens/LoadingScreen', 'Loading')
  return (
    <View style={[tailwind('bg-white items-center justify-center flex-1 h-full w-full relative')]}>
      <Spinner
        color='#5b10ff'
        overlayColor='rgba(255, 255, 255, 1)'
        visible
        textContent={loadingMessage}
      />
    </View>
  )
}
