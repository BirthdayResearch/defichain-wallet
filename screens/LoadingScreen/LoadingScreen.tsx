import * as React from 'react'
import Spinner from 'react-native-loading-spinner-overlay'
import tailwind from 'tailwind-rn'
import { View } from '../../components/Themed'
import { translate } from '../../translations'
import { StyleSheet } from 'react-native'

interface LoadingScreenProps {
  message?: string
}

export default function LoadingScreen (props: LoadingScreenProps): JSX.Element {
  const { message } = props
  const loadingMessage = message ?? translate('screens/LoadingScreen', 'Loading')
  return (
    <View style={[styles.loaderContainer, tailwind('bg-white')]}>
      <Spinner
        color='#5b10ff'
        overlayColor='rgba(255, 255, 255, 1)'
        visible
        textContent={loadingMessage}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  loaderContainer: {
    height: '100%',
    width: '100%',
    position: 'relative',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
})
