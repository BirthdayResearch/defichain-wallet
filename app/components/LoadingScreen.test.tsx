import { render } from '@testing-library/react-native'
import * as React from 'react'

import LoadingScreen from './LoadingScreen'

jest.mock('../contexts/ThemeProvider')
it('<LoadingScreen/> should display default text', () => {
  const { getByText } = render(<LoadingScreen />)

  expect(getByText('Loading')).toBeTruthy()
})

it('<LoadingScreen/> should display passed text', () => {
  const { getByText } = render(<LoadingScreen message='Creating Wallet' />)

  expect(getByText('Creating Wallet')).toBeTruthy()
})

it('<LoadingScreen /> should match snapshot', () => {
  const tree = render(<LoadingScreen message='Creating Wallet' />).toJSON()
  expect(tree).toMatchSnapshot()
})
