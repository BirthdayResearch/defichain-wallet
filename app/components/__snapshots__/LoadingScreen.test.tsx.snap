// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "height": "100%",
      "justifyContent": "center",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#5b10ff"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                undefined,
              ]
            }
          >
            Creating Wallet
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
